### PR TITLE
Request payload may be consumed to early with multipart config

### PR DESCRIPTION
**Pull request by matthiaso**
372439

### CODE REVIEW

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.ui.html.json;
```
Comment (matthiaso) [inline org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Can this be right? Wanted to place it near `UploadRequestHandler`

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  protected void removeConfigIfNotNeeded(ServletRequest req0) {
+    if (!(req0 instanceof HttpServletRequest)) {
+      return;
+    }
+
+    HttpServletRequest req = (HttpServletRequest) req0;
+
+    if (!needsConfig(req)) {
+      // see org.eclipse.jetty.server.Request.__MULTIPART_CONFIG_ELEMENT (literal value)
```
Comment (matthiaso) [inline org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Maybe place somewhere else where we have jetty classes accessible? What about `org.eclipse.scout.rt.jetty` module?

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java (hunk 1 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Works for me. Near UploadRequestHandler is correct, the only caller of this part of handler are the ScoutJS widgets using Scout json protocol -> package ui.,html.json matches.

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      req.removeAttribute("org.eclipse.jetty.multipartConfig");
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Remove config here or add it explicitly only for /upload Path here and remove in default UiServlet registration?
I personally would prefer to remove the global multipart config and just add it here for file upload paths.

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java (hunk 2 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Placing in rt.jetty requires to copy the fileupload-pattern -> I would leave this class here and copy the jetty constant, adding a unit test to enforce contant equality (and adding jetty in unit test as dependency)

Comment (matthiaso) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Add constant in this class

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java (hunk 1 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Moved to org.eclipse.scout.rt.ui.html.app where we have access to both constants (Jetty and `UploadRequestHandler`).

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java (hunk 2 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Moved to org.eclipse.scout.rt.ui.html.app where we have access to both constants (Jetty and `UploadRequestHandler`).

File: org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java (hunk 3 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java]: Changed behavior from remove to add.

File: org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+
+/**
+ * <p>
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java]: Remove, only use <p> to separate paragraphs, not using </p> too.

File: org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * We must not add this multipart config to all requests (to the servlet holder) as otherwise reading parameters may
+ * consume the request payload (to determine parameters) which makes other body reading methods e.g.
+ * {@link HttpServletRequest#getInputStream()} unusable.
+ * </p>
+ *
+ * @see <a href="https://github.com/jetty/jetty.project/issues/1175">jetty/jetty.project issue #1175</a>
+ * @see HttpServletRequest#getParameter(String)
+ * @see Request#extractContentParameters()
+ */
+public class MultipartConfigFilter implements Filter {
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java]: Rename to UiServletMultipartConfigFilter (according to property)?

File: org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  @Override
+  public void doFilter(ServletRequest req0, ServletResponse resp, FilterChain chain) throws IOException, ServletException {
+    addConfigIfNeeded(req0);
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java]: addConfigIfNeeded -> addIfMultipartConfigRequired

File: org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    HttpServletRequest req = (HttpServletRequest) req0;
+
+    if (needsConfig(req)) {
+      req.setAttribute(Request.__MULTIPART_CONFIG_ELEMENT, CONFIG.getPropertyValue(UiServletMultipartConfigProperty.class));
+    }
+  }
+
+  protected boolean needsConfig(HttpServletRequest request) {
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/filter/MultipartConfigFilter.java]: isMultipartConfigRequired

File: org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/UiServletContributors.java [hunk 8]
```diff
@@ -80,6 +78,15 @@ public void contribute(ServletContextHandler handler) {
     }
   }
 
+  @Order(4000)
+  public static class MultipartConfigFilterContributor implements IServletFilterContributor {
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.ui.html.app/src/main/java/org/eclipse/scout/rt/ui/html/app/UiServletContributors.java]: UiServletMultipartConfigFilterContributor
@Order(100)
JavaDoc stating that usually multipart config is registered by servletHolder.getRegistration().setMultipartConfig

**Review by stephan-merkli — APPROVED** [review]

**Pull request merged**